packages/@react-aria/dnd/src/useDrag.ts

```diff
@@ -304,6 +297,14 @@ export function useDrag(options: DragOptions): DragResult {
           e.stopPropagation();
           startDragging(e.target as HTMLElement);
         }
+      },
+      onClick(e) {
+        // Handle NVDA/JAWS in browse mode, and touch screen readers. In this case, no keyboard events are fired.
+        if (isVirtualClick(e.nativeEvent) || modalityOnPointerDown.current === 'virtual') {
+          e.preventDefault();
+          e.stopPropagation();
+          startDragging(e.target as HTMLElement);
```

Comment:
If we're already dragging, clicking the drag target should cancel dragging. In this case, the drag operation will cancel and then immediately start dragging again.